languages/ruby/outline.scm

```diff
@@ -1,67 +1,182 @@
+; Class definitions, e.g. `class Foo`
 (class
     "class" @context
     name: (_) @name) @item
 
+; Singleton class definitions `class << self`
 (singleton_class
     "class" @context
     "<<" @context
     value: (self) @context
 ) @item
 
+; Method definition with a modifier, e.g. `private def foo`
 (body_statement
-    ((identifier) @context
-    (#match? @context "^(private|protected|public)$")) @item
-)
-
-(body_statement
-  (call
-      method: (identifier) @context
-      arguments: (argument_list
-          (method
-              "def" @context
-              name: (_) @name)
-          )) @item
+    (call
+        method: (identifier) @context
+        arguments: (argument_list
+            (method
+                "def" @context
+                name: (_) @name)
+            )) @item
 )
 
+; Method definition without modieifer, e.g. `def foo`
 (body_statement
     (method
         "def" @context
         name: (_) @name) @item
 )
 
+; Root method definition with modifier, e.g. `private def foo`
 (program
-  (call
-      method: (identifier) @context
-      arguments: (argument_list
-          (method
-              "def" @context
-              name: (_) @name)
-          )) @item
+    (call
+        method: (identifier) @context
+        arguments: (argument_list
+            (method
+                "def" @context
+                name: (_) @name)
+            )) @item
 )
 
+; Root method definition without modifier, e.g. `def foo`
 (program
     (method
         "def" @context
         name: (_) @name) @item
 )
 
-(singleton_method
-    "def" @context
-    object: (_) @context
-    "." @context
-    name: (_) @name) @item
+; Root singleton method definition, e.g. `def self.foo`
```

Comment:
There’s no modified root singleton method definition, because I don’t think anyone uses that. It would be easy enough to add if we find we do need it later.